pa11ycrawler/pipelines.py

```diff
@@ -60,6 +62,51 @@ def process_item(self, item, spider):  # pylint: disable=unused-argument
             return item
 
 
+def ignore_rules_for_url(spider, url):
+    """
+    Returns a list of ignore rules from the given spider,
+    that are relevant to the given URL.
+    """
+    ignore_rules = getattr(spider, "pa11y_ignore_rules", {}) or {}
+    return itertools.chain.from_iterable(
+        rule_list
+        for url_glob, rule_list
+        in ignore_rules.items()
+        if fnmatch.fnmatch(url, url_glob)
+    )
+
+
+def ignore_rule_matches_result(ignore_rule, pa11y_result):
+    """
+    Returns a boolean result of whether the given ignore rule matches
+    the given pa11y result. The rule only matches the result if *all*
+    attributes of the rule match.
+    """
+    return all(
+        fnmatch.fnmatch(pa11y_result.get(attr), ignore_rule.get(attr))
+        for attr in ignore_rule.keys()
+    )
+
+
+def load_pa11y_results(stdout, spider, url):
+    """
+    Load output from pa11y, filtering out the ignored messages.
+    The `stdout` parameter is a bytestring, not a unicode string.
+    """
+    if not stdout:
+        return []
+
+    results = json.loads(stdout.decode('utf8'))
+
+    ignore_rules = ignore_rules_for_url(spider, url)
+    for rule in ignore_rules:
```

Comment:
Yes, that would probably be computationally faster. However, having the flexibility to specify URL _patterns_ instead of fixed URLs is a very useful feature, and makes it easier to understand the intended scope of each ignore rule. As a result, we use this `ignore_rules_for_url` to determine the list of rules that match the given URL.